### DISCUSSION TITLE
Next release date?

### DISCUSSION THREAD

**Discussion by gaveitatry2** [violentmonkey/violentmonkey#1193] [General]
Just out of curiosity, no hurry, but when do you think the current Beta version will become official?  I ask because the word-wrap broke after 2.12.4 and I was forced to use the old version for like 9 months (I don't remember how long exactly) until 2.12.9 finally came out.  So I was amped up and ready to use 2.12.9 for the working word-wrap, but then there were some new script breaking bugs that you have already fixed in Beta.  So I'm using the Beta version now, but is it going to be another 9 months before 2.12.10 comes out?  Again, no pressure, but here is my idea.  When word-wrap broke in 2.12.4, instead of waiting 9 months to release a version that has working word-wrap, you should have just added word-wrap right away and released it as a new version right immediately.  So 2.12.9 would have only had the word-wrap fixed.  Then you could have spent the next 9 months adding other stuff that you needed to add and call that 2.12.10.  Then when you figured out that there were a few script breaking bugs in this newest version (that caused my old waitForKeyElement scripts to not load on first try), you could have added that fix right away and called it 2.12.11.  But the fix still hasn't been added yet, so I'm still stuck on either Beta or 2.12.4.  And then you can spend the next 9 months after this on on the other stuff that you are currently working on now that doesn't involve those aforementioned bug fixes.  It just makes sense to get the biggest fixes out of the way, the things that are crucial and matter a lot to end users.  And the other stuff that is also important but not as crucial and can be overlooked, work on that after you get the biggest fixes released right away.

**Comment by gaveitatry2** [comment]
Oh one more thing.  I wasn't going to ask when this Beta version was going to become official.  Or write the other stuff.  But I noticed problems with the Beta version.  For instance, I can't turn a userscript off with the toggle button.  I can't turn a userscript off by toggling all userscripts off.  The only way to get a script to stop working is to go to Extensions page and temporarily disable the addon.  So since Beta doesn't have the full features that an official version does, it would be nice to finally be on an official version that has all the important bug fixes after not being on the latest up-to-date official version for 9 months (and however more months before this Beta becomes official).

**Reply by tophf** [reply]
> can't turn a userscript off with the toggle button. I can't turn a userscript off by toggling all userscripts off.

I can't reproduce it in the latest beta 2.12.9.5 with the latest Firefox or Chrome.

> since Beta doesn't have the full features that an official version does,

What do you mean specifically? There should be no difference.

**Reply by gaveitatry2** [reply]
I'll try to reproduce it later via gif or video.  Btw, I was on 2.12.9.4 Beta, and it won't update to 2.12.9.5 Beta.  When I check for updates on my extensions page, it says no updates available.  I thought you said that "as of 2.12.9.4, it will update automatically".  Oh well.

**Reply by gera2ld** [reply]
Sorry, I was wrong about that, the auto update should work since v2.12.9.5 because of https://github.com/violentmonkey/violentmonkey/commit/b630d974497b809f267baffbc1a746cff045e28d

**Reply by gaveitatry2** [reply]
So I did a quick check, and the Beta version isn't acting up anymore.  I think that my browser was just acting wonky and I needed to restart the browser before the extension would behave properly again.  Sorry about that.

**Reply by gaveitatry2** [reply]
I am thinking that ViolentMonkey BETA is not updating correctly.  (At least not now, since 2.12.9.9.)  I had version 2.12.9.9.  (I still do.)  Last night, a popup on the bottom right of my laptop screen (above the system tray) said something about a ViolentMonkey update error.  It was on my other laptop, and I didn't have time to read the full message.  But anyway, just now, on my main laptop I checked and it is still showing version 2.12.9.9.  Even after I "check for updates".  But your GitHub Releases page says that the newest version is 2.12.10.  Maybe it will eventually update on it's own in a couple of days?  I don't know.

**Reply by gaveitatry2** [reply]
Oh nevermind.  It's the official release.  I can't wait.  I've been wanting to be on the official release for a long time, but I was stuck with BETA because of some bugs.  Thank you.  I'll update my extensions tomorrow after I've gone to bed.

**Comment by tophf** [comment]
There are no estimates or hard schedules.
